beeai/supervisor/qe_data.py

```diff
@@ -0,0 +1,43 @@
+import asyncio
+from functools import cache
+from pydantic import BaseModel
+
+from .http_utils import aiohttp_session
+
+
+class TestLocationInfo(BaseModel):
+    component: str
+    qa_contact: str
+    tests_location: str | None = None
+    test_config_location: str | None = None
+    test_trigger_method: str | None = None
+    test_result_location: str | None = None
+    test_docs_url: str | None = None
+    notes: str | None = None
+
+
+async def fetch_qe_data_map() -> dict[str, dict[str, dict[str, str]]]:
+    url = "https://gitlab.cee.redhat.com/otaylor/jotnar-qe-data/-/raw/main/jotnar_qe_data.json?ref_type=heads&inline=false"
+    async with aiohttp_session().get(url) as response:
+        response.raise_for_status()
+        return await response.json(content_type=None)
+
+
+@cache
+def get_qe_data_map_task() -> asyncio.Task[dict[str, dict[str, dict[str, str]]]]:
+    return asyncio.create_task(fetch_qe_data_map())
```

Comment:
we never clear the cache right? data map can get stale? should probably `get_qe_data_map_task.clear_cache()` in a timeout or something...